### PR TITLE
Update authentication documentation

### LINKED ISSUE #1386: clarification on handling auth/access to storage from GKE
It is not clear what is meant by "we handle authentication for you with no configuration" interms of implementation.  Does this mean the environment variables are already present and ready to use?
```
        Google::Cloud::Storage.new(
          project: ENV['GOOGLE_CLOUD_PROJECT'],
          keyfile: File.absolute_path(ENV['GOOGLE_CLOUD_KEYFILE'])
        )
```

or that the library will access them directly from the environment so they do not need to be passed in?
```
Google::Cloud::Storage.new()
```

### PR DESCRIPTION
**Pull request by quartzmo**
[closes #1386]

### CODE REVIEW

**Comment by quartzmo** [comment]
This PR is currently incomplete. Please let me know if the proposed changes to google-cloud-bigquery look good. Once approved, I will copy these changes to other packages and update the PR.

File: google-cloud-bigquery/lib/google/cloud/bigquery.rb [hunk 1]
```diff
@@ -27,11 +27,12 @@ module Cloud
     # BigQuery?](https://cloud.google.com/bigquery/what-is-bigquery).
     #
     # The goal of google-cloud is to provide an API that is comfortable to
-    # Rubyists. Authentication is handled by {Google::Cloud#bigquery}. You can
-    # provide the project and credential information to connect to the BigQuery
-    # service, or if you are running on Google Compute Engine this configuration
-    # is taken care of for you. You can read more about the options for
-    # connecting in the [Authentication
+    # Rubyists. Your authentication credentials are detected automatically in
```
Comment (blowmage) [inline google-cloud-bigquery/lib/google/cloud/bigquery.rb]: Does this change need to be applied to any other gems?

Comment (quartzmo) [reply google-cloud-bigquery/lib/google/cloud/bigquery.rb]: Yes. But how does it look?

> Please let me know if the proposed changes to google-cloud-bigquery look good. Once approved, I will copy these changes to other packages and update the PR.

Comment (blowmage) [reply google-cloud-bigquery/lib/google/cloud/bigquery.rb]: Oops, I somehow missed your comment on the issue before I commented. Looks great to me. :+1:

**Pull request merged**